src/connection.ts

```diff
@@ -39,7 +39,10 @@ export const SFDX_HTTP_HEADERS = {
 };
 
 export const DNS_ERROR_NAME = 'Domain Not Found';
-const DNS_TIMEOUT = Math.max(3, new Env().getNumber('SFDX_DNS_TIMEOUT') || 3);
+// Timeout for DNS lookup polling should default to 30 seconds
+const DNS_TIMEOUT = new Env().getNumber('SFDX_DNS_TIMEOUT', 30) as number;
+// Retry frequency for DNS lookup polling should be at least 10 seconds
```

Comment:
Pete has strong feelings about boundary conditions on this.  I'll let him comment

Reply:
this PR has context: https://github.com/forcedotcom/sfdx-core/pull/362

Reply:
and what's taking 30 seconds?  something about the bamboo proxy caching?

Reply:
@mshanemc updated

Reply:
@mdonnalley Not sure what affect negative numbers have, but its simple to eliminate those using Math.max. Also give one the ability to enforce a minimum.